lib/grandpa/grandpa.go

```diff
@@ -193,8 +193,13 @@ func (s *Service) Start() error {
 	s.tracker.start()
 
 	go func() {
-		if err := s.initiate(); err != nil {
-			logger.Criticalf("failed to initiate: %s", err)
+		for {
+			// TODO: sometimes grandpa fails to initiate due to a "Key not found"
+			// error, this shouldn't happen.
+			if err := s.initiate(); err != nil {
+				logger.Criticalf("failed to initiate: %s", err)
```

Comment:
Panic? Error channel? 🤔

Reply:
I think this is the same sort of issue as with BABE, we probably need to add an error channel so the higher-level service handler can either panic or restart the service. I'll make an issue for it

Reply:
#2235